Remotion/Data/DomainObjects.UnitTests/Database/DataDomainObjects_Patches.sql

```diff
@@ -0,0 +1,533 @@
+CREATE TABLE [TableInheritance_BaseClassWithInvalidRelationClassIDColumns] (
+  [ID] uniqueidentifier NOT NULL,
+  [ClassID] varchar (100) NOT NULL,
+  [Timestamp] rowversion NOT NULL,
+  
+  [ClientID] uniqueidentifier NULL, -- Note: To conform to mapping column ClientIDClassID must not be defined.
+  [ClientIDClassID] varchar (100) NULL, 
+  [DomainBaseID] uniqueidentifier NULL, -- Note: To conform to mapping column DomainBaseIDClassID must be defined.
+  
+  [DomainBaseWithInvalidClassIDValueID] uniqueidentifier NULL, 
+  [DomainBaseWithInvalidClassIDValueIDClassID] varchar (100) NULL, 
+  [DomainBaseWithInvalidClassIDNullValueID] uniqueidentifier NULL, 
+  [DomainBaseWithInvalidClassIDNullValueIDClassID] varchar (100) NULL, 
+  
+  -- Note: This table does not need to have foreign keys, because rows cannot be read because of invalid ClassID column structure
+  CONSTRAINT [PK_TableInheritance_BaseClassWithInvalidRelationClassIDColumns] PRIMARY KEY CLUSTERED ([ID])
+) 
+GO
+
+-- CustomDataType_ClassWithCustomDataType
+IF EXISTS (SELECT * FROM INFORMATION_SCHEMA.Views WHERE TABLE_NAME = 'CustomDataType_ClassWithCustomDataTypeView' AND TABLE_SCHEMA = 'dbo')
+  DROP VIEW [dbo].[CustomDataType_ClassWithCustomDataTypeView]
+GO
+
+IF EXISTS (SELECT * FROM INFORMATION_SCHEMA.Tables WHERE TABLE_NAME = 'CustomDataType_ClassWithCustomDataType')
+  DROP TABLE [dbo].[CustomDataType_ClassWithCustomDataType]
+GO
+
+CREATE TABLE [dbo].[CustomDataType_ClassWithCustomDataType]
+(
+  [ID] uniqueidentifier NOT NULL,
+  [ClassID] varchar (100) NOT NULL,
+  [Timestamp] rowversion NOT NULL,
+  [CompoundDataTypeValueStringValue] nvarchar (100) NULL,
+  [CompoundDataTypeValueInt32Value] int NULL,
+  [SimpleDataTypeValue] nvarchar (max) NULL,
+  CONSTRAINT [PK_ClassWithCustomDataType] PRIMARY KEY CLUSTERED ([ID])
+)
+GO
+
+CREATE VIEW [dbo].[CustomDataType_ClassWithCustomDataTypeView] ([ID], [ClassID], [Timestamp], [CompoundDataTypeValueStringValue], [CompoundDataTypeValueInt32Value], [SimpleDataTypeValue])
+  WITH SCHEMABINDING AS
+  SELECT [ID], [ClassID], [Timestamp], [CompoundDataTypeValueStringValue], [CompoundDataTypeValueInt32Value], [SimpleDataTypeValue]
+    FROM [dbo].[CustomDataType_ClassWithCustomDataType]
+  WITH CHECK OPTION
+GO
+
+-- ==============================================
+--  2. PROCEDURES
+-- ==============================================
+
+-- rpf_testSPQuery
+IF OBJECT_ID ('rpf_testSPQuery', 'P') IS NOT NULL 
+  DROP PROCEDURE rpf_testSPQuery;
+GO
+
+CREATE PROCEDURE rpf_testSPQuery
+AS
+  SELECT * FROM [Order] WHERE [OrderNo] = 1 OR [OrderNo] = 3 ORDER BY [OrderNo] ASC
+GO
+
+-- rpf_testSPQueryWithParameter
+IF OBJECT_ID ('rpf_testSPQueryWithParameter', 'P') IS NOT NULL 
+  DROP PROCEDURE rpf_testSPQueryWithParameter;
+GO
+
+CREATE PROCEDURE rpf_testSPQueryWithParameter
+  @customerID uniqueidentifier
+AS
+  SELECT * FROM [Order] WHERE [CustomerID] = @customerID ORDER BY [OrderNo] ASC
+GO
+
+-- ==============================================
```

Comment:
What's the deal with these types? Why do we have to do those manually?

Reply:
Those are located in the default storage provider, which we did not include in the automatic schema generation

Reply:
I think that's not by design. we should fix this instead of doing a patch since they're perfectly fine c# types that result in tables/views

Reply:
They should be moved into the correct storage provider? Or should we include the default storage provider?

Reply:
Don't know if adding the default provider or adding a new provider group mapped to the same storage provider or simply moving those types to the testdomain provider group is the best choice. I probably would go with the last choice first and see where it leads.